app/models/manageiq/providers/google/cloud_manager/refresh_parser.rb

```diff
@@ -236,6 +238,20 @@ def parse_parent_image(instance)
         parent_image_uid
       end
 
+      def get_unique_flavors(flavors)
+        unique_flavors = []
+        flavor_ids     = []
+
+        flavors.each do |flavor|
+          unless flavor_ids.include?(flavor.id.to_i)
+            unique_flavors << flavor
+            flavor_ids     << flavor.id.to_i
+          end
+        end
+
+        unique_flavors
+      end
+
```

Comment:
Though actually, just `.uniq` / `.distinct` should be fine... it's a relation, right?